cmd/system-probe/app/compile.go

```diff
@@ -0,0 +1,102 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package app
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/DataDog/datadog-agent/cmd/system-probe/api/module"
+	spconfig "github.com/DataDog/datadog-agent/cmd/system-probe/config"
+	ddconfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/runtime"
+	rc "github.com/DataDog/datadog-agent/pkg/runtimecompiler"
+	"github.com/DataDog/datadog-agent/pkg/runtimecompiler/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/DataDog/datadog-go/v5/statsd"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	statsdPoolSize = 64
+)
+
+func init() {
+	SysprobeCmd.AddCommand(compileCommand)
+}
+
+var compileCommand = &cobra.Command{
+	Use:   "compile",
+	Short: "Compile eBPF programs",
+	Long:  "Perform runtime compilation of eBPF programs used by the system-probe",
+	Run:   runtimeCompile,
+}
+
+func runtimeCompile(_ *cobra.Command, _ []string) {
+	/*
+		Because we expect the compile command to be used in a kubernetes init container prior to the
+		system-probe container, this function should never return an error, else it could prevent
+		the system-probe from being able to start.
```

Comment:
The concern here is that any container wouldn't start right, including the `agent`? And `system-probe` would start with an error presumably if compilation fails?

Reply:
It would be an init container on the system-probe container, so I believe if the init container fails then the agent container would be okay, but the system-probe container would fail

Reply:
init containers are pod-level, so in a k8s environment, that would run first and then the other containers in the pod will run.

Reply:
ah, yep you're right. An error in that container would prevent all the other ones from starting.

Reply:
Do we know how long it takes to compile everything? I worry about adding a delay to all the other containers starting.